src/app/services/game-play.service.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+
+import { GamePlayService } from './game-play.service';
+import { AnswerService } from './answer.service';
+
+describe('GamePlayService', () => {
+  let service: GamePlayService;
+  let guessResult: string;
+  let acceptedGuessesResult: string[];
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+      providers: [AnswerService]
+    });
+    service = TestBed.inject(GamePlayService);
+    guessResult = '';
+    acceptedGuessesResult = [];
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  })
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should return A when A is added to currentGuess', () => {
+    let newLetter = 'A';
+
+    service.addLetter(newLetter);
+    service.currentGuess$.subscribe(currentGuess => guessResult = currentGuess);
+
+    expect(guessResult.includes(newLetter)).toBeTruthy()
+  });
+
+  it('should return ABCDE when A, B, C, D and E are added to currentGuess', () => {
+    let letters = 'ABCDE';
+
+    for (let i = 0; i < 5; i++) service.addLetter(letters[i]);
+    service.currentGuess$.subscribe(currentGuess => guessResult = currentGuess);
+
+    expect(guessResult.includes(letters)).toBeTruthy()
+  });
+
+  it('should return DORIT when DORITE is attempted for currentGuess', () => {
+    let letters = 'DORITE';
+
+    for (let i = 0; i < 6; i++) service.addLetter(letters[i]);
+    service.currentGuess$.subscribe(currentGuess => guessResult = currentGuess);
+    let canDoriteBeGuessed = 
+
+    expect(guessResult.includes('DORIT')).toBeTruthy()
+  });
+
+  it('should return LOL when removeLetter is called 3 times to currentGuess', () => {
+    let letters = 'LOLLY';
+
+    for (let i = 0; i < 5; i++) service.addLetter(letters[i]);
+    for (let i = 0; i < 2; i++) service.removeLetter();
+    service.currentGuess$.subscribe(currentGuess => guessResult = currentGuess);
+
+    expect(guessResult.includes('LOL')).toBeTruthy()
+  });
+
+  it('should submit a guess when valid guess is submitted', () => {
+    let validGuess = 'ADEPT';
+    let guessIdx = 0;
+    let currentGuess = '';
+
+    for (let i = 0; i < 5; i++) service.addLetter(validGuess[i]);
+    service.submitGuess();
+    service.acceptedGuesses$.subscribe(guesses => acceptedGuessesResult = guesses);
```

Comment:
An easier way to do this is to use [firstValueFrom](https://rxjs.dev/deprecations/to-promise#firstvaluefrom)

```Typescript
const acceptedGuessesResult = await firstValueFrom(service.acceptedGuesses$)
```

Reply:
In fact, this can all be consolidated down to a one-liner:

```Typescript
expect((await firstValueFrom(service.acceptedGuesses$)).includes(validGuess)).toBeTruthy()
```

I would consider changing from using `.includes` to something like `.toEqual` instead. It's just a stricter check than `.includes`:

```Typescript
expect(await firstValueFrom(service.acceptedGuesses$)).toEqual(validGuess)
```